elements/lisk-transactions/src/register_multisignature_account.ts

```diff
@@ -58,71 +66,109 @@ const validateInputs = ({
 	}
 
 	if (
-		!isValidInteger(lifetime) ||
-		lifetime < MULTISIGNATURE_MIN_LIFETIME ||
-		lifetime > MULTISIGNATURE_MAX_LIFETIME
+		!isValidInteger(numberOfSignatures) ||
+		numberOfSignatures < MIN_NUMBER_OF_SIGNATURES ||
+		numberOfSignatures > MAX_NUMBER_OF_SIGNATURES
 	) {
 		throw new Error(
-			`Please provide a valid lifetime value. Expected integer between ${MULTISIGNATURE_MIN_LIFETIME} and ${MULTISIGNATURE_MAX_LIFETIME}.`,
+			`Please provide a valid numberOfSignatures value. Expected integer between ${MIN_NUMBER_OF_SIGNATURES} and ${MAX_NUMBER_OF_SIGNATURES}.`,
+		);
+	}
+
+	if (mandatoryPublicKeys.length > numberOfSignatures) {
+		throw new Error(
+			'The numberOfSignatures should be more than or equal to the number of mandatory passphrases.',
 		);
 	}
 
 	if (
-		!isValidInteger(minimum) ||
-		minimum < MULTISIGNATURE_MIN_KEYSGROUP ||
-		minimum > MULTISIGNATURE_MAX_KEYSGROUP
+		numberOfSignatures >
+		mandatoryPublicKeys.length + optionalPublicKeys.length
 	) {
 		throw new Error(
-			`Please provide a valid minimum value. Expected integer between ${MULTISIGNATURE_MIN_KEYSGROUP} and ${MULTISIGNATURE_MAX_KEYSGROUP}.`,
+			`Please provide a valid numberOfSignatures. numberOfSignatures (${numberOfSignatures}) is bigger than the count of optional (${optionalPublicKeys.length}) and mandatory (${mandatoryPublicKeys.length}) keys.`,
 		);
 	}
 
-	if (keysgroup.length < minimum) {
+	if (
+		mandatoryPublicKeys.length + optionalPublicKeys.length >
+			MAX_NUMBER_OF_KEYS ||
+		mandatoryPublicKeys.length + optionalPublicKeys.length < MIN_NUMBER_OF_KEYS
+	) {
 		throw new Error(
-			'Minimum number of signatures is larger than the number of keys in the keysgroup.',
+			`Please provide a valid number of mandatory and optional keys. Expected integer between ${MIN_NUMBER_OF_SIGNATURES} and ${MAX_NUMBER_OF_SIGNATURES}.`,
 		);
 	}
 
-	validateKeysgroup(
-		keysgroup,
-		MULTISIGNATURE_MIN_KEYSGROUP,
-		MULTISIGNATURE_MAX_KEYSGROUP,
-	);
+	// Check key duplication between sets
+	const repatedKeys = findRepeatedKeys(optionalPublicKeys, mandatoryPublicKeys);
```

Comment:
for below 3, can we just simply check 
```
const uniqueKeys = Array.from(new Set([...mandatoryPublicKeys, ...optionalPublicKeys]));
uniqueKeys.length !== mandatoryPublicKeys.length + optionalPublicKeys.length?
```

Reply:
Change though we can no longer indicate if the problem is in mandatory or optional so single message is shown.